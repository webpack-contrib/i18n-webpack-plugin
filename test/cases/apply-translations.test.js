import processFile from '../cases.setup';

describe('apply-translations', () => {
  let translated;

  beforeAll(() => {
    const translations = {
      'static-key': 'translated static key',
      '{count} nights': [
        '{count} ночь',
        '{count} ночи',
        '{count} ночей',
      ],
    };

    return processFile('apply-translations.code.js', translations, { pluralRuleNumber: 7 })
      .then(({ file }) => {
        translated = require.requireActual(file);
      });
  });

  it('should return translated keys', () => {
    expect(translated).toMatchSnapshot();
  });
});
