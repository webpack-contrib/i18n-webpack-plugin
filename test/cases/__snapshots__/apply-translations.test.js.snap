// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apply-translations should return translated keys 1`] = `
Object {
  "defaults": "default value",
  "missingKey": "missing-key2",
  "parametrizedKey": "env: dev",
  "staticKey": "translated static key",
  "templateKey": "5 ночей",
}
`;
